src/components/AnalysisPage/ThicknessAnalysisBox/ThickAnalysisBox.jsx

```diff
@@ -57,21 +57,48 @@ const ThickAnalysisBox = ({
       setSelectedThreshold(Number(value));
       setShowCheckmark(true);
       setAnalysisComplete(true);
+      checkForOrDownloadPly(data);
     }
   };
 
-  async function loadPLYFromIndexedDB(partId, units, thresholdValue) {
+  const downloadPlyFilePlaceIn3dViewer = (
+    jobId,
+    partId,
+    units,
+    thresholdValue
+  ) => {
+    fetch(`/api/v0/thickness/${jobId}/visual/ply`)
+      .then((res) => res.blob())
+      .then((blob) => {
+        setFileNameForUpload("file.ply");
+        const file = new File([blob], "file.ply", { type: blob.type });
+        setFileFor3dModel(file);
+
+        storeBlob(partId, units, thresholdValue, blob);
+      });
+  };
+
+  const checkForOrDownloadPly = async (data) => {
     try {
-      const blob = await retrieveBlob(partId, units, thresholdValue);
+      const blob = await retrieveBlob(data.part_id, data.units, data.threshold);
       if (blob) {
         setFileNameForUpload("file.ply");
         const file = new File([blob], "file.ply", { type: blob.type });
         setFileFor3dModel(file);
+      } else {
+        if (listOfThicknessData.find((data) => data.id === data.id)) {
+          downloadPlyFilePlaceIn3dViewer(
+            data.id,
+            data.part_id,
+            data.units,
+            data.threshold
+          );
+        }
       }
     } catch (error) {
       console.error("Failed to load PLY from IndexedDB:", error);
```

Comment:
We may want to add an error popup in the UI if the download fails rather than just a log

Reply:
We took care of this in downloadPlyToViewer.js in a new commit below!